src/fides/api/service/connectors/sql_connector.py

```diff
@@ -189,14 +190,28 @@ def retrieve_data(
 
         logger.info("Starting data retrieval for {}", node.address)
         with client.connect() as connection:
-            self.set_schema(connection)
-            if (
-                query_config.partitioning
-            ):  # only BigQuery supports partitioning, for now
-                return self.partitioned_retrieval(query_config, connection, stmt)
-
-            results = connection.execute(stmt)
-            return self.cursor_result_to_rows(results)
+            try:
+                self.set_schema(connection)
+                if (
+                    query_config.partitioning
+                ):  # only BigQuery supports partitioning, for now
+                    return self.partitioned_retrieval(query_config, connection, stmt)
+
+                results = connection.execute(stmt)
+                return self.cursor_result_to_rows(results)
+            except Exception as exc:
+                # Check if table exists using qualified table name
+                if not self.table_exists(connection, node):
+                    # Central decision point - will raise TableNotFound or ConnectionException
+                    qualified_table_name = self.get_qualified_table_name(node)
+                    self.handle_table_not_found(
+                        node=node,
+                        table_name=qualified_table_name,
+                        operation_context="data retrieval",
+                        original_exception=exc,
+                    )
+                # Table exists or can't check - re-raise original exception
+                raise
```

Comment:
Central handling for missing tables. Instead of checking for table existence before each query, we check after any exception. The other way to check for missing tables would be to parse various exceptions for missing tables but that seemed too fragile.